### PR TITLE
Rust clippy172 v1

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
None

Describe changes:
- Fix clippy 1.72 warnings

@jasonish how can we prevent CI from going red when clippy 1.73 will be out ?

### CODE REVIEW

**Comment by jasonish** [comment]
> @jasonish how can we prevent CI from going red when clippy 1.73 will be out ?

We could pin clippy to a version, and making updating it a manual process.

**Comment by catenacyber** [comment]
Clippy nightly does not give the newt warnings somehow ?

**Comment by jasonish** [comment]
> Clippy nightly does not give the newt warnings somehow ?

I don't think we should run nightly for linting.  By definition lints that it produces may not be fixable without suppression in our MSRV, or could change from night to night as they realize mistakes were made.

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .flow.spare | 1951025 | 2162602 | 110.84% | 

Pipeline 15787

**Comment by catenacyber** [comment]
> We could pin clippy to a version, and making updating it a manual process.

Not perfect either...

**Review by victorjulien — APPROVED** [review]

**Comment by catenacyber** [comment]
Jason has also one commit to upgrade the crate bit flags cf https://github.com/OISF/suricata/pull/9421/commits/6b6c598e47f49f863088a824ca34dd5b403da367

File: rust/src/detect/iprep.rs [hunk 1]
```diff
@@ -84,7 +84,7 @@ pub fn detect_parse_iprep(i: &str) -> IResult<&str, DetectIPRepData> {
     let (i, name) = take_while(is_alphanumeric_or_slash)(i)?;
     // copy as to have final zero
     let namez = CString::new(name).unwrap();
-    let cat = unsafe { SRepCatGetByShortname(namez.as_ptr() as *const i8) };
+    let cat = unsafe { SRepCatGetByShortname(namez.as_ptr()) };
```
Comment (victorjulien) [inline rust/src/detect/iprep.rs:87]: This gives errors on my arm builders:
```
   Compiling suricata v7.0.1-dev (/builds/inliniac/suricata-ci/suricata/rust)
error[E0308]: mismatched types
  --> src/detect/iprep.rs:87:46
   |
87 |     let cat = unsafe { SRepCatGetByShortname(namez.as_ptr()) };
   |                        --------------------- ^^^^^^^^^^^^^^ expected `*const i8`, found `*const u8`
   |                        |
   |                        arguments to this function are incorrect
   |
   = note: expected raw pointer `*const i8`
              found raw pointer `*const u8`
note: function defined here
  --> src/detect/iprep.rs:74:12
   |
74 |     pub fn SRepCatGetByShortname(name: *const i8) -> u8;
   |            ^^^^^^^^^^^^^^^^^^^^^
For more information about this error, try `rustc --explain E0308`.
error: could not compile `suricata` (lib) due to previous error


   Compiling suricata v7.0.1-dev (/builds/inliniac/suricata-ci/suricata/rust)
error[E0308]: mismatched types
  --> src/detect/iprep.rs:87:46
   |
87 |     let cat = unsafe { SRepCatGetByShortname(namez.as_ptr()) };
   |                        --------------------- ^^^^^^^^^^^^^^ expected `i8`, found `u8`
   |                        |
   |                        arguments to this function are incorrect
   |
   = note: expected raw pointer `*const i8`
              found raw pointer `*const u8`
note: function defined here
  --> src/detect/iprep.rs:74:12
   |
74 |     pub fn SRepCatGetByShortname(name: *const i8) -> u8;
   |            ^^^^^^^^^^^^^^^^^^^^^
For more information about this error, try `rustc --explain E0308`.
error: could not compile `suricata` due to previous error
```
Not sure why the 2 errors are different, perhaps rustc version differences.

Comment (victorjulien) [reply rust/src/detect/iprep.rs:87]: IIRC C type `char` is unsigned in arm.

Comment (catenacyber) [reply rust/src/detect/iprep.rs:87]: Indeed

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/9433

**Pull request closed without merging**